### PR TITLE
De-duplicate code in JwtHelper derived classes

### PR DESCRIPTION
**Pull request by ctron**
This change does pull up common code from JwtHelper derived classes back into JwtHelper.

### CODE REVIEW

File: core/src/main/java/org/eclipse/hono/util/JwtHelper.java [hunk 1]
```diff
@@ -200,4 +206,62 @@ public Key resolveSigningKey(JwsHeader header, Claims claims) {
...
+        Objects.requireNonNull(sharedSecret);
+        Objects.requireNonNull(instanceSupplier);
+
+        final T result = instanceSupplier.get();
+        result.setSharedSecret(getBytes(sharedSecret));
+        result.tokenLifetime = Duration.ofSeconds(tokenExpiration);
+        return result;
+    }
+
+    protected static <T extends JwtHelper> T forSigning(final String forWhat,
```
Comment (sophokles73) [inline core/src/main/java/org/eclipse/hono/util/JwtHelper.java]: I like the idea of pulling up the common code using the `Supplier` to return the specific sub-class type :+1: 
However, I do not really like the *forWhat* parameter which seems to be there only to pass a fragment of a log statement. FMPOV we can simply omit it and leave out the corresponding information in the log statements, i.e. I can also live with a simple
`using shared secret [34 bytes] for signing tokens`

Comment (ctron) [reply core/src/main/java/org/eclipse/hono/util/JwtHelper.java]: Ok, I will update the PR.

Comment (ctron) [reply core/src/main/java/org/eclipse/hono/util/JwtHelper.java]: PR is updated.

**Comment by sophokles73** [comment]
Thanks for contributing :-)

**Pull request merged**